sisl/viz/plots/pdos.py

```diff
@@ -411,10 +411,12 @@ def _read_from_H(self, kgrid, kgrid_displ, Erange, nE, E0, distribution):
 
             PDOS.append(spin_PDOS)
 
-        if not self.H.spin.is_diagonal:
+        if len(spin_indices) == 1:
             PDOS = PDOS[0]
+        else:
+            PDOS = np.array(PDOS)
```

Comment:
Isn't PDOS here an array of shape `(2, 1, no, nE)` in the polarized case?

Reply:
no, it shouldn't. Or did I do something wrong. It should be `(2, no, nE)`

Reply:
Doesn't `eigenstate.PDOS` return `(1, no, nE)`? And then if you append one after the other you get `(2, 1, no, nE)`

Reply:
oh my, you are correct... I didn't catch that ;)

Reply:
I think the `_read_from_wfsx` method also needs some tweak to reshape the output of `eigenstate.PDOS` into `(no, nE)`, I'm not sure the sum `(no, nE)` + `(1, no, nE)` -> `(no, nE)` is appropiately handled :sweat_smile:

Reply:
in `numpy` (*, x, y) + (x, y) -> (*, x, y)` using broadcasting rules. So if `*` is 1 then all is fine, otherwise `(x, y)` would be added to each element of `*`

Reply:
I think changing 

```python
if nspin == 4:
    spin = slice(None)
```
to
```python
if nspin != 2:
    spin = slice(None)
```
should do

Reply:
> in numpy (, x, y) + (x, y) -> (, x, y)using broadcasting rules. So ifis 1 then all is fine, otherwise(x, y)would be added to each element of

Yes, but it's the opposite: (x, y) + (1, x, y) -> (x, y). is it also handled correctly?

Reply:
yes